src/modules/cpus/avr/avr5.zig

```diff
@@ -24,39 +25,63 @@ pub inline fn cbi(comptime reg: u5, comptime bit: u3) void {
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-            \\ jmp _unhandled_vector
-        );
+pub const vector_table = blk: {
+    std.debug.assert(std.mem.eql(u8, "RESET", std.meta.fields(microzig.chip.VectorTable)[0].name));
+    var asm_str: []const u8 = "jmp _start\n";
+
+    const has_interrupts = @hasDecl(microzig.app, "interrupts");
+    if (has_interrupts) {
+        if (@hasDecl(microzig.app.interrupts, "RESET"))
+            @compileError("Not allowed to overload the reset vector");
+
+        inline for (std.meta.declarations(microzig.app.interrupts)) |decl| {
+            if (!@hasField(microzig.chip.VectorTable, decl.name)) {
+                var msg: []const u8 = "There is no such interrupt as '" ++ decl.name ++ "'. ISRs the 'interrupts' namespace must be one of:\n";
+                inline for (std.meta.fields(microzig.chip.VectorTable)) |field| {
+                    if (!std.mem.eql(u8, "RESET", field.name)) {
+                        msg = msg ++ "    " ++ field.name ++ "\n";
+                    }
+                }
+
+                @compileError(msg);
+            }
+        }
     }
 
+    inline for (std.meta.fields(microzig.chip.VectorTable)[1..]) |field| {
+        const new_insn = if (has_interrupts) overload: {
+            if (@hasDecl(microzig.app.interrupts, field.name)) {
+                const handler = @field(microzig.app.interrupts, field.name);
+                const calling_convention = switch (@typeInfo(@TypeOf(@field(microzig.app.interrupts, field.name)))) {
+                    .Fn => |info| info.calling_convention,
+                    else => @compileError("Declarations in 'interrupts' namespace must all be functions. '" ++ field.name ++ "' is not a function"),
+                };
+
+                const exported_fn = switch (calling_convention) {
+                    .Unspecified => struct {
+                        fn wrapper() callconv(.C) void {
+                            if (calling_convention == .Unspecified) // TODO: workaround for some weird stage1 bug
+                                @call(.{ .modifier = .always_inline }, handler, .{});
+                        }
+                    }.wrapper,
+                    else => @compileError("Just leave interrupt handlers with an unspecified calling convention"),
+                };
+
+                const options = .{ .name = field.name, .linkage = .Strong };
```

Comment:
I would probably add an `_isr_` prefix to the symbol name here, and we need to make sure the symbol is located directly after the vector table (same for `_start` :thinking: )